test/test_tabel.py

```diff
@@ -186,22 +187,23 @@ def test_replace_column(self):
 
 
 class TestSaveNRead(object):
-    def test_save_and_read(self, tbls, tmpdir):
-        for tbl in tbls:
-            if len(tbl) > 0:
-                tbl.save(os.path.join(str(tmpdir), "test.csv"), fmt="csv")
-                tbl_b = read_tabel(os.path.join(str(tmpdir), "test.csv"), fmt="csv")
-                assert tbl_b.valid
-                # tbl.save(tmpdir+"/test.gz", fmt="gz")
-                # tbl_b = read_tabel(tmpdir+"/test.gz", fmt="gz")
-                # assert tbl_b.valid
-                tbl.save(os.path.join(str(tmpdir), "test.npz"), fmt="npz")
-                tbl_b = read_tabel(os.path.join(str(tmpdir), "test.npz"), fmt="npz")
-                assert set(tbl.columns) == set(tbl_b.columns)
-                for c in tbl.columns:
-                    assert np.all(tbl[:,c] == tbl_b[:,c])
-
-
+    def test_save_n_read(self, tbls, tmpdir):
+        for header in [True, False]:
+            for fmt in ['csv', 'gz', 'npz']:
+                for tbl in tbls:
+                    if len(tbl) > 0:
+                        fn = os.path.join(str(tmpdir), "test."+fmt)
+                        tbl.save(fn, header=header)
+                        tbl_r = read_tabel(fn, header=header)
+                        assert (set(tbl_r.columns) == set(tbl.columns)) or (not header), (tbl, tbl_r)
+                        assert len(tbl_r) == len(tbl), (tbl, tbl_r)
+                        assert tbl_r.valid, (tbl, tbl_r)
+
+                        if not tbl_r.columns[0].isdigit() and len(tbl)>2:
+                            # header sniffer should work
+                            tbl_r = read_tabel(fn, header=None)
+                            assert set(tbl_r.columns) == set(tbl.columns), (tbl, tbl_r)
+                            assert len(tbl_r) == len(tbl), (tbl, tbl_r)
```

Comment:
Oh I see what happened, I think an older version got merged in a newer one or something, I'll manually take your test code and 'reject' the merge.